drivers/lsm6dsl/lsm6dsl.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @file
+ * @brief       Device driver implementation for the LSM6DSL 3D accelerometer/gyroscope.
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ */
+
+#include "xtimer.h"
+
+#include "lsm6dsl.h"
+#include "lsm6dsl-internal.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+
+int lsm6dsl_init(lsm6dsl_t *dev, const lsm6dsl_params_t *params)
+{
+    uint8_t res = 0;
+
+    assert(dev && params);
+
+    dev->params = *params;
+
+    i2c_acquire(dev->params.i2c);
+    i2c_init_master(dev->params.i2c, I2C_SPEED_NORMAL);
+
+    /* Reboot */
+    res = i2c_write_reg(dev->params.i2c, dev->params.addr,
+                        LSM6DSL_REG_CTRL3_C, LSM6DSL_CTRL3_C_BOOT);
+
+    xtimer_usleep(5000);
+
+    i2c_read_reg(dev->params.i2c, dev->params.addr, LSM6DSL_REG_WHO_AM_I, &res);
+    if (res != LSM6DSL_WHO_AM_I) {
+        i2c_release(dev->params.i2c);
+        return -1;
+    }
+
+    /* Set acc odr / full scale */
+    res = i2c_write_reg(dev->params.i2c, dev->params.addr, LSM6DSL_REG_CTRL1_XL,
+                        ((dev->params.acc_odr << LSM6DSL_CTRL_ODR_SHIFT) |
+                         (dev->params.acc_fs << LSM6DSL_CTRL_FS_SHIFT)));
+    /* Set gyro odr / full scale */
+    res = i2c_write_reg(dev->params.i2c, dev->params.addr, LSM6DSL_REG_CTRL2_G,
+                        ((dev->params.gyro_odr << LSM6DSL_CTRL_ODR_SHIFT) |
+                         (dev->params.gyro_fs << LSM6DSL_CTRL_FS_SHIFT)));
+
+    /* Set continuous mode */
+    uint8_t fifo_odr = MAX(dev->params.acc_odr, dev->params.gyro_odr);
+    res = i2c_write_reg(dev->params.i2c, dev->params.addr, LSM6DSL_REG_FIFO_CTRL5,
+                        (fifo_odr << LSM6DSL_FIFO_CTRL5_FIFO_ODR_SHIFT) |
+                        LSM6DSL_FIFO_CTRL5_CONTINUOUS_MODE);
+    res = i2c_write_reg(dev->params.i2c, dev->params.addr, LSM6DSL_REG_FIFO_CTRL3,
```

Comment:
same with this one because of another reason:
`drivers/lsm6dsl/lsm6dsl.c:64: style (unreadVariable): Variable 'res' is assigned a value that is never used`

The `make static-test` returns interesting results :)